airflow/providers/google/cloud/secrets/secret_manager.py

```diff
@@ -101,9 +106,17 @@ def __init__(
                     "`connections_prefix`, `variables_prefix` and `sep` should "
                     f"follows that pattern {SECRET_ID_PATTERN}"
                 )
-        self.credentials, self.project_id = get_credentials_and_project_id(
-            keyfile_dict=gcp_keyfile_dict, key_path=gcp_key_path, scopes=gcp_scopes
-        )
+        try:
+            self.credentials, self.project_id = get_credentials_and_project_id(
+                keyfile_dict=gcp_keyfile_dict, key_path=gcp_key_path, scopes=gcp_scopes
+            )
+        except (DefaultCredentialsError, FileNotFoundError):
+            log.exception(
+                'Unable to load credentials for GCP Secret Manager. '
+                'Make sure that the keyfile path, dictionary, or GOOGLE_APPLICATION_CREDENTIALS '
+                'environment variable is correct and properly configured.'
```

Comment:
I wonder if it’d be a good idea to list the paths and check if `GOOGLE_APPLICATION_CREDENTIALS` is set (probably not a good idea to show its value). That’d make debugging much easier.

Reply:
The `log.exception` will probably show that error as that is raised from/by `get_credentials_and_project_id` _I think_

https://github.com/apache/airflow/blob/866a601b76e219b3c043e1dbbc8fb22300866351/airflow/providers/google/cloud/utils/credentials_provider.py#L225

https://github.com/apache/airflow/blob/866a601b76e219b3c043e1dbbc8fb22300866351/airflow/providers/google/cloud/utils/credentials_provider.py#L287-L288

https://github.com/apache/airflow/blob/866a601b76e219b3c043e1dbbc8fb22300866351/airflow/providers/google/cloud/utils/credentials_provider.py#L276-L278

Reply:
Yes, but I was thinking it’d use useful to know what the value of `self.key_path` is in the above context, so I don’t need to hunt the value in my configs. But maybe google-auth already includes that in the exception, no idea.